src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -102,10 +102,9 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                     }
                     isPrecompiledFunctionApp = isPrecompiledFunctionApp || functionMetadata.Language == DotNetScriptTypes.DotNetAssembly;
                 }
-
-                isDotnetIsolatedApp = IsDotnetIsolatedApp(functionMetadataCollection, SystemEnvironment.Instance);
             }
 
+            bool isDotnetIsolatedApp = IsDotnetIsolatedApp(functionMetadataCollection, SystemEnvironment.Instance);
```

Comment:
So what was the actual issue here that caused the bug I logged? The main thing you're changing seems to be that you'll now be consulting the function metadata in non-bundle cases as well.

Reply:
For most of dotnet-isolated cases the extension bundle will not configured. In those cases, we would report isDotnetIsolated as false even though that is not the case. With the fix we would hydrate regardless of the bundles configuration.

Reply:
So the issue was that we would incorrectly report isDotnetIsolated as false if the extension bundle is not configured.